src/computational_graph/graph.jl

```diff
@@ -260,28 +260,31 @@ julia> g = feynman_diagram([𝑓⁺(1)𝑓⁻(2)𝜙(3), 𝑓⁺(4)𝑓⁻(5)
 
 julia> g.subgraphs
 3-element Vector{Graph{Float64, Float64}}:
- 2: propagtor graph from f⁻(5)f⁺(1)
- 3: propagtor graph from f⁻(2)f⁺(4)
- 4: propagtor graph from ϕ(3)ϕ(6)
+ 2: propagator graph from f⁻(5)f⁺(1)
+ 3: propagator graph from f⁻(2)f⁺(4)
+ 4: propagator graph from ϕ(3)ϕ(6)
 ```
 """
 function feynman_diagram(vertices::Vector{OperatorProduct}, topology::Vector{Vector{Int}};
     external::Union{Nothing,AbstractVector}=nothing, factor=one(_dtype.factor), weight=zero(_dtype.weight), name="", type=:generic)
 
     operators = [o for v in vertices for o in v.operators]
-    permutation = collect(Iterators.flatten(topology))
+    contraction = collect(Iterators.flatten(topology))
     if isnothing(external)
-        external = [i for i in eachindex(operators) if i ∉ permutation]
+        external = [i for i in eachindex(operators) if i ∉ contraction]
     end
-    @assert length(unique(permutation)) == length(permutation) # no repeated index
+    @assert length(unique(contraction)) == length(contraction) # no repeated index
     @assert length(unique(external)) == length(external) # no repeated index
-    @assert Set(union(external, permutation)) == Set(eachindex(operators)) # external + permutation must exhaust all operators
+    @assert Set(union(external, contraction)) == Set(eachindex(operators)) # external + permutation must exhaust all operators
+
+    permutation = union(contraction, external)
+    _external = intersect(external, contraction)
```

Comment:
See comments in Slack thread. We should remove this line, since `_external == external` for all valid use cases. Better to check the input and throw an ArgumentError if the user specifies invalid external indices.